### ISSUE TITLE
[SCREENREADER] Calendar Widget - Fieldset needs to be a sibling to the button trigger  for screen reader usability

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#2709]
## Description

The staging versions of our request-date and select-date pickers have the `<fieldset>` of radios / checkboxes appended to the end of row. This is creating a situation where JAWS and VoiceOver are ignoring the inputs unless users are navigating by using the `TAB` key. Screen reader users who navigate by virtual cursor are missing the inputs, and will not be able to use the calendar effectively.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

**As a screen reader user:**
* I want to have clear instructions for use read out. This could be accomplished by adding an SR-only paragraph block to the table's `aria-describedby` H2.
* I want to be able to expand a cell by clicking or pressing the `<button>`, and navigate to the form inputs immediately by:
  * Pressing `Ctrl + Opt + RIGHT_ARROW` or `TAB` in VoiceOver
  * Pressing `DOWN_ARROW` or `f` or `TAB` in NVDA
  * Pressing `f` or `TAB` in JAWS

## Environment

* Windows 10, MacOS Mojave
* IE11, Chrome, Firefox, Safari
* JAWS, NVDA, VoiceOver

## Possible Fixes (optional)

The form elements have to be moved inside table cells. I've mocked a very simple version of this on Codepen: https://codepen.io/tpierce_402/pen/LYYxLxN . This mockup has been tested with the big three screen readers on desktop, and works well for all.

~The codepen will be updated to include `aria-controls` attributes ASAP. This has been tested with the big three screen readers for efficacy.~

## WCAG or Vendor Guidance (optional)

* [Keyboard: Understanding SC 2.1.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/keyboard-operation-keyboard-operable.html)
* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)
* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

## Screenshots or Trace Logs

![Screen Shot 2019-10-24 at 9.33.15 AM.png](https://images.zenhubusercontent.com/5ac217b74b5806bc2bcd3fc8/5dc992d4-d454-4619-a25e-9d4ee3c6e299)

---
![Screen Shot 2019-10-24 at 9.32.42 AM.png](https://images.zenhubusercontent.com/5ac217b74b5806bc2bcd3fc8/e63a3d03-0859-417b-bc55-4594a775cfcc)

**Comment by 1Copenut** [comment]
@jbalboni I retested the calendar widget on staging. Used NVDA, JAWS, and VoiceOver with their respective browser pairings.

None of the screen readers were handling the new cell properly. I'm pretty sure it has to do with injecting another cell into the row, which is confusing either the screen reader or the browser, which in turn, confuses the screen reader.

I did some dabbling with CSS in the browser, and came up with a treatment where the `div` containing our checkboxes or radios was inlined and made vertical. It reads a little better to me, and should support the radio behavior, but I'll defer to you, Mark, and any designers on the project if this is a good departure.

LMK if we should have a quick discussion meeting and I'll put it on the calendar.

/cc @rtwell 

---
![Screen Shot 2020-01-07 at 3 18 07 PM](https://user-images.githubusercontent.com/934879/71930582-96370b80-3161-11ea-8001-39c7a522ced5.png)

**Comment by jbalboni** [comment]
Thanks @1Copenut! Yeah, it was a fun solution, but I can see why it would have trouble. I think your suggestion there is workable for the AM/PM version of the calendar, but there's another version that has 2+ timeslots and wouldn't work to stack them vertically (there might be like 6 boxes on top of each other).

I think there are two main avenues for fixing this:
1. Keep the current design, have the FE put the timeslot row inside the cell for the day and position it absolutely. This can work and won't break the cell layout, but we'll have to calculate the height to figure out how much to push later rows down. This is kinda gross because we have to do it on resizes, too. But it's doable.
2. Tweak the design so that we don't have to move the rows around when the times are being shown (i.e. have the times overlay the calendar days, like a modal). This doesn't seem great, but maybe @rtwell has some ideas that make this route better than 1).

**Comment by 1Copenut** [comment]
Pinging @CrystabelReiter for VSP design coverage. My apologies for not handling this sooner Crystabel!

**Comment by jbalboni** [comment]
I put up a PR for option 1 here: https://github.com/department-of-veterans-affairs/vets-website/pull/11438.

**Comment by 1Copenut** [comment]
Re-tested on Sunday, January 19 and the latest changes have made this behave properly in VoiceOver. Will retest in NVDA and JAWS when changes are on staging.

**Comment by lenaecb** [comment]
@1Copenut - as we prep for our MVP launch readiness of VAOS, I want to confirm that this needs to be addressed before we start any phased roll out of the new VAOs tool. Thanks!

**Comment by 1Copenut** [comment]
@lenaecb Yes, ideally we have this closed out before launch. I'll retest the screen readers (JAWS in particular) against the calendar widget in the morning and see how we're faring on this ticket. @jbalboni has done a lot of refactoring already, and it's possible we've got this one closed out. Will update tomorrow.

**Comment by 1Copenut** [comment]
@lenaecb I've retested this on JAWS and it's still not behaving as I hoped. I'm going to keep this issue open and step it down to a priority 2 item. I'm also writing a few related tickets that will help bridge the gap for JAWS users.

I'll send this as a question to my contact at the VA 508 office with our plan early next week.

**Comment by jbalboni** [comment]
@1Copenut Do you want to keep this issue open? Not sure what to do with it right now.

**Comment by 1Copenut** [comment]
@jbalboni Let's go ahead and close it. We solved the underlying issue through other refactors, IMO.
